apps/console/src/features/roles/components/wizard-updated/role-permissions/role-permissions.tsx

```diff
@@ -0,0 +1,278 @@
+            error: apiResourcsListFetchRequestError,
+            mutate: mutateAPIResourcesListFetchRequest
+        } = useAPIResourcesList(apiResourceSearchQuery);
+
+        const {
+            data: selectedAPIResource,
+            isLoading: isSelectedAPIResourceFetchRequestLoading,
+            isValidating: isSelectedAPIResourceFetchRequestValidating,
+            error: selectedAPIResourceFetchRequestError        
+        } = useAPIResourceDetails(selectedAPIResourceId);
+
+        /**
+         * Show error if user list fetch request failed
+         */ 
+        useEffect(() => {
+            if ( selectedAPIResourceFetchRequestError ||  apiResourcsListFetchRequestError) {
+                dispatch(
+                    addAlert({
+                        description: t("console:manage.features.roles.addRoleWizard.forms.rolePermission." +
+                            "notifications.fetchAPIResourceError.error.description"),
+                        level: AlertLevels.ERROR,
+                        message: t("console:manage.features.roles.addRoleWizard.forms.rolePermission." +
+                            "notifications.fetchAPIResourceError.error.message")
+                    })
+                );
+                setIsPermissionStepNextButtonDisabled(true);
+            } else {
+                setIsPermissionStepNextButtonDisabled(false);
+            }
+        }, [ selectedAPIResourceFetchRequestError, apiResourcsListFetchRequestError ]);
+
+        useEffect(() => {
+            const options: DropdownProps[] = [];
+    
+            apiResourcesList?.apiResources?.map((apiResource: APIResourceInterface) => {
+                if (!selectedAPIResources.find((selectedAPIResource: APIResourceInterface) => 
+                    selectedAPIResource?.id === apiResource?.id)) {
+                    options.push({
+                        key: apiResource.id,
+                        text: apiResource.name,
+                        value: apiResource.id
+                    });
+                }
+            });
+        
+            setAPIResourcesListOptions(options);
+        }, [ apiResourcesList, selectedAPIResources ]);
+
+        /**
+         * Add API resource to the selected API resources list.
+         */
+        useEffect(() => {
+            if (selectedAPIResource 
+                && !isSelectedAPIResourceFetchRequestLoading 
+                && !isSelectedAPIResourceFetchRequestValidating) {
+                
+                if (!selectedAPIResources.find(
+                    (apiResource: APIResourceInterface) => selectedAPIResource?.id === apiResource?.id)) {
+                    setSelectedAPIResources([ selectedAPIResource, ...selectedAPIResources ]);
+                }
+
+                selectedAPIResourceId && setSelectedAPIResourceId(undefined);
+            }
+        }, [ selectedAPIResource ]);
+
+        // TODO: Need to handle the error state.
```

Comment:
Is this intentional? is there a blocker to handle the error state?

Reply:
Error state is handled, will remove the comment

Reply:
Fixed with https://github.com/wso2/identity-apps/pull/4278/commits/dc1ed8886e3e85f06247f47bbcf6eae95045a9bd